tests/model/test_crystal_model.py

```diff
@@ -596,3 +596,26 @@ def test_set_scan_varying_B_covariance(crystal_class):
         assert cov_B_at_scan_point == cov_B_2d
         cell_sd_at_scan_point = xl.get_cell_parameter_sd_at_scan_point(i)
         assert cell_sd_at_scan_point == pytest.approx(cell_sd)
+
+
+def test_recalculated_cell():
+    xl = Crystal(
+        real_space_a=(10, 0, 0),
+        real_space_b=(0, 11, 0),
+        real_space_c=(0, 0, 12),
+        space_group_symbol="P 1",
+    )
+
+    assert not xl.has_recalc_unit_cell()
+
+    uc1 = xl.get_unit_cell()
+    xl.set_recalc_unit_cell(uc1)
```

Comment:
Might it not be better to set a _different_ recalculated unit cell, to ensure that the stored unit cell and recalculated unit cells are actually different object?